pkg/graveler/ref/manager_test.go

```diff
@@ -1249,3 +1259,90 @@ func TestManager_DeleteExpiredImports(t *testing.T) {
 	require.NoError(t, it.Err())
 	require.Equal(t, 2, count)
 }
+
+func TestManager_GetRepositoryMetadata(t *testing.T) {
+	ctx := context.Background()
+	r, _ := testRefManager(t)
+	const (
+		repoID = "repo1"
+	)
+	repository, err := r.CreateRepository(context.Background(), repoID, graveler.Repository{
+		StorageNamespace: "s3://",
+		CreationDate:     time.Now(),
+		DefaultBranchID:  "main",
+	})
+	testutil.Must(t, err)
+
+	t.Run("get_on_non_existing_repo", func(t *testing.T) {
+		_, err := r.GetRepositoryMetadata(ctx, "not_exist")
+		require.ErrorIs(t, err, graveler.ErrNotFound)
+	})
+
+	t.Run("basic", func(t *testing.T) {
+		expected := &graveler.RepositoryMetadata{}
+		metadata, err := r.GetRepositoryMetadata(ctx, repository.RepositoryID)
+		require.NoError(t, err)
+		require.Equal(t, metadata, expected)
+	})
+}
+
+func TestManager_SetRepositoryMetadata(t *testing.T) {
```

Comment:
TODO: add a multithreaded test.  It will not always catch errors, but maybe we'll get lucky with the race detector a few times?

Reply:
We don't test the retry mechanism in this scope - these are ref manager tests

Reply:
Sounds good.  So where can we test this? (OK to open as a tech-debt issue, but I usually feel safer running a few rounds of a concurrent test.)

Reply:
I will add this as part of #6068